app/src/main/java/org/oppia/app/walkthrough/end/WalkthroughFinalFragmentPresenter.kt

```diff
@@ -3,25 +3,84 @@ package org.oppia.app.walkthrough.end
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.Transformations
+import org.oppia.app.R
 import org.oppia.app.databinding.WalkthroughFinalFragmentBinding
 import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.ProfileId
+import org.oppia.app.model.Topic
+import org.oppia.app.walkthrough.WalkthroughActivity
+import org.oppia.domain.topic.TopicController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
 import javax.inject.Inject
 
 /** The presenter for [WalkthroughFinalFragment]. */
 @FragmentScope
 class WalkthroughFinalFragmentPresenter @Inject constructor(
-  private val fragment: Fragment
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val logger: Logger,
+  private val topicController: TopicController
 ) {
   private lateinit var binding: WalkthroughFinalFragmentBinding
+  private lateinit var walkthroughFinalViewModel: WalkthroughFinalViewModel
+  private lateinit var topicId: String
+  private lateinit var profileId: ProfileId
+  private lateinit var topicName: String
 
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?, topicId: String): View? {
     binding = WalkthroughFinalFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+    this.topicId = topicId
+    val internalProfileId = activity.intent.getIntExtra(WalkthroughActivity.WALKTHROUGH_ACTIVITY_INTERNAL_PROFILE_ID_KEY, 1)
```

Comment:
Default value should be `-1`

Reply:
@rt4914 how to check for a smaller device because the minimum I have right now is a 5-inch phone

Reply:
Possibly you should create an AVD (Android Virtual Device) from Tools in android studio and if that does not work out than while writing test cases, before performing any check or click on view, just add one `scrollTo` action so that test cases work on smaller devices too. You can see such test cases in various files where we have written code for landscape.

Reply:
okay got it will add scroll view and add a scrollto action inorder run the test isCompleletyDisplayed

Reply:
done